src/python/tests/tod.py

```diff
@@ -133,3 +135,46 @@ def test_local_signal(self):
 
         stop = MPI.Wtime()
         elapsed = stop - start
+
+class TestApplyGain(MPITestCase):
+
+    def setUp(self):
+        self.gain = {
+                "TIME":np.arange(10),
+                "CH-1":2*np.ones(10, dtype=np.float32),
+                "CH-2":3*np.ones(10, dtype=np.float32)
+                }
+        self.gain["CH-2"][5:] = 0
+    
+    def test_write_calibration_file(self):
+        write_calibration_file("test_cal.fits", gain)
+
+    def test_op_applygain(self):
+
+        self.outdir = "toast_test_output"
+        if self.comm.rank == 0:
+            if not os.path.isdir(self.outdir):
+                os.mkdir(self.outdir)
+        self.mapdir = os.path.join(self.outdir, "dipole")
+        if self.comm.rank == 0:
+            if not os.path.isdir(self.mapdir):
+                os.mkdir(self.mapdir)
+
+        # Note: self.comm is set by the test infrastructure
+
+        self.toastcomm = Comm(world=self.comm)
+        self.data = Data(self.toastcomm)
+
+        self.detnames = ['bore']
+        self.dets = {
+            'CH-1' : np.array([0.0, 0.0, 1.0, 0.0]),
+            'CH-2' : np.array([0.0, 1.0, 0.0, 0.0])
+            }
+
+        # generate input data of ones for each of the 2 channels and 10 timestamps
+        # from 1 to 10, no need for pointing
+
```

Comment:
@tskisner I need to generate fake data of just ones for the 2 channels, what is the easiest way to do that? I don't need pointing information.

Reply:
thanks, I postponed creating `tod/applygain.py` because I don't remember how you add a new file to the build system. can you please remind me?

Reply:
also, with `TODCache` I get this error:

```
Traceback (most recent call last):
  File "/global/common/software/pico/zonca/toast_prefix_cori/lib/python3.6/site-packages/toast/tod/tod.py", line 1037, in __del__
    self.cache.clear()
AttributeError: 'TODCache' object has no attribute 'cache'
```

Reply:
Just add the file to the Makefile.am in that directory (see where the other files are listed).  Then re-run autogen.sh, configure, etc.  I have not seen that error in the destructor before...  Having said that, I see that both the TOD base class and the derived TODCache class are trying to clear the cache in their destructors.  It should not harm anything to do that, but we only really need that in the base class destructor.

Reply:
thanks, another thing, what is the key in the TODCache created by default by write? I tried with "signal" but didn't work, then I tried to track it down into the code but couldn't find it.